lib/rubocop/target_finder.rb

```diff
@@ -100,6 +96,27 @@ def find_files(base_dir, flags)
       Dir.glob(pattern, flags).select { |path| FileTest.file?(path) }
     end
 
+    private
+
+    def ls_git_files(base_dir)
+      return unless Dir.exist?("#{base_dir}/.git")
+
+      `git ls-files #{base_dir} --exclude-standard \
```

Comment:
I think we avoid to use backtick to execute a command, because it executes a command with a shell. Shell is aware of meta characters such as `*`. If `base_dir` has meta characters, this command will show unexpected behaviour.
I think we can use `Open3` instead.

Reply:
I don't think that `Open3` is better:

```
pry: main > `echo $SHELL`
=> "/usr/bin/fish\n"
pry: main > output, status = Open3.capture2('echo $SHELL')
=> ["/usr/bin/fish\n", #<Process::Status: pid 9644 exit 0>]
```

Are there any more arguments?

Reply:
Sorry, my suggestion was bad.
Open3 executes a command with shell if it receives a command that includes spaces. You should separate the argument to avoid shell.


For example

```ruby
require 'open3'

# with a shell
Open3.capture3('echo $SHELL') # => ["/bin/zsh\n", "", #<Process::Status: pid 21026 exit 0>]

# without shell
Open3.capture3('echo', '$SHELL') # => ["$SHELL\n", "", #<Process::Status: pid 21041 exit 0>]
```

Reply:
> Open3 executes a command with shell if it receives a command that includes spaces. You should separate the argument to avoid shell.

What? Why? You just execute to commands: `echo` (empty) and `$SHELL` (it runs the shell).

`Open3` is just for capturing stderr, stdout and status. That's all. I don't see another difference from backticks.

Reply:
It does not execute two command, it's only one command. `Open3.capture3('echo', '$SHELL')` is almost same with `Open3.capture3("echo '$SHELL'")`. But the later executes `echo` command with a shell.

> Open3 is just for capturing stderr, stdout and status. That's all. I don't see another difference from backticks.

We cannot separate command and arguments explicitly if we use backticks. But we can do that with `Open3`, `system()` and so on.

You can see the documentation for `Kernel.spawn` method. `Open3` method passes received arguments to `spawn`, so `spawn` receives the same arguments with `Open3`.

https://docs.ruby-lang.org/en/2.4.0/Process.html#method-c-spawn

>  command...:
>   commandline                 : command line string which is passed to the standard shell
>   cmdname, arg1, ...          : command name and one or more arguments (This form does not use the shell. See below for caveats.)
>   [cmdname, argv0], arg1, ... : command name, argv[0] and zero or more arguments (no shell)


`Open3.capture3('echo', '$SHELL')` is the second style.

Reply:
I'm still not sure about `Open3` for globs (`*`), especially how to test it (it works with backticks for me), but I switched to `Open3` for another feature: if `base_dir` is nested dir of git-repository, and sub-dir, obviously, doesn't have `.git` directory, but `git` executable works correctly. So, I'm capturing error output now, and check the status.